### ISSUE TITLE
Prettier Latex Block Math in Markdown: adds extra backslash

### ISSUE THREAD

**Issue by anishmittal2020** [prettier/prettier#8571]
**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEASVAdD8AeMAEAwgPoBs+WuBAEscAIwBMAvhdnHvgPJmr4DU+VKUocYwLszos+mDACcAlgHMAFjACG8+RADuskezzBCk6cz6Dho49SnAWN8ZMtCnwBHHnKAnhdQgADQgEAAOMIrQAM7IoFo6ugAKWggxKBoANroaPjHBAEbyGmAA1nAwAMqhxYpQysgw8gCucMHqALYZAOqqivBR1WBwFal9igBufT7I4FF5ILVRXjCJRcrtGsgAZplLwQBWUTgAQkWl5RUa7XAAMrVw27utIIc4FbXKGXAAik0Q8I8MnsQNV5Et5DMNvISgATPRQIIgpSwLqKGEwVTIAAcAAZgqEdEsukVQjMCXBweMHsEAI5-eCrMJpEAaKIAWigcDgMO5iPkcDpin5qw0602SB2QOeS3aigazWlHy+v3+DwlT2CmnyqPRmKQjE1RUUGQ+hAg7Q2MwpAFZEU0lgAVDT5NKS4HjFoASSgPNgFTASnCAEEfRUYD4voClsxmEA)
```sh
--parser markdown
```

**Input:**
```markdown
$$\text C_6 \text H_{12} \text O_6$ + $6\text{O}_{2}$ $\rightarrow$ $6\text{CO}_{2}$ + $6\text{H}_{2}\text{O}$ + $\text{energy}$$
```

**Output:**
```markdown
\$$\text C_6 \text H_{12} \text O_6$ + $6\text{O}_{2}$ $\rightarrow$ $6\text{CO}_{2}$ + $6\text{H}_{2}\text{O}$ + $\text{energy}$\$

```

**Expected behavior:**
See Input

**Comment by anishmittal2020** [comment]
@thorn0  Prettier adds extra backslash before the last and the first dollar sign!

**Comment by thorn0** [comment]
Fixed by #7938
